Core/Core/Use Cases/UploadManager.swift

```diff
@@ -152,14 +152,16 @@ open class UploadManager: NSObject, URLSessionDelegate, URLSessionTaskDelegate,
                 file.id = nil
                 file.bytesSent = 0
                 try context.save()
-                let body = PostFileUploadTargetRequest.Body(name: url.lastPathComponent, on_duplicate: .rename, parent_folder_id: nil, size: file.size)
+                let body = PostFileUploadTargetRequest.Body(name: url.lastPathComponent, on_duplicate: .rename, parent_folder_path: folderPath, size: file.size)
                 let request = PostFileUploadTargetRequest(context: uploadContext, body: body)
                 environment.api.makeRequest(request) { response, _, error in
                     self.context.performAndWait {
                         defer { callback?() }
-                        guard let target = response, error == nil, let file = try? self.context.existingObject(with: objectID) as? File else {
-                            self.sendFailedNotification()
-                            return
+                        guard let file = try? self.context.existingObject(with: objectID) as? File else {
+                            return self.sendFailedNotification()
+                        }
+                        guard let target = response, error == nil else {
+                            return self.complete(file: file, error: error)
```

Comment:
I ran into issues with exceeding the quota erroring in this target request. That error wasn't getting propagated to the `uploadError` property.